hibernate-core/src/main/java/org/hibernate/query/hql/internal/QuerySplitter.java

```diff
@@ -183,6 +188,10 @@ private static SqmRoot<?> findUnmappedPolymorphicReference(SqmDeleteOrUpdateStat
 		return null;
 	}
 
+	public interface FromCopyProvider {
```

Comment:
This reference is only ever used internally right?  Assuming so, an interface is probably overkill

Reply:
Oops, no I see now you expose use it in other classes

Reply:
Andrea and I talked about this today, and we came to the conclusion that this can be done differently in a non layer breaking manner.